### ISSUE TITLE
Storage Foundation migration

### ISSUE THREAD

**Issue by hcldan** [WICG/file-system-access#330]
I'm not sure where to put this.  I've been following Storage Foundation project for a while, as well as Native File System.
I have seen talk about convergence, and I agree that this would be a good thing.  It seems Google has been flailing around in attempts to bring file system support to browsers, and some consolidation and solidity in these efforts is welcome.

I would like to know how the process of deprecating the other APIs (those in or out of origin-trials) will be done.
Can you please be more transparent about direction? I understand that you may still be working out what the direction will be, but it's hard for a company to see what's going on right now and know what horse to bet on.

**Comment by fivedots** [comment]
Hey Dan,

Our current focus is on [Access Handles](https://github.com/WICG/file-system-access/blob/main/AccessHandle.md) as the next iteration (and replacement) of Storage Foundation. While there might be some extensions to the Storage Foundation origin trial (currently running until 1 week before Chrome 96 reaches stable), it is unlikely that that particular surface will be available after the trial expires.

There is a new Origin Trial happening for Sync Access Handles running from Chrome 95 to 98, you can register [here](https://developer.chrome.com/origintrials/#/view_trial/3378825620434714625) if you'd like to try it out. While we are still receiving feedback and can make changes, we generally expect to ship something  similar to what is in described in the proposal/exposed in the trial.

I'll add a note on the Storage Foundation explainer to make this clearer. I hope this answers your question!

**Comment by hcldan** [comment]
This thrashing of apis and lack of communication is costing us a pretty significant amount of time and resources trying to see where google is going and what they will eventually support....

**Comment by tomayac** [comment]
@hcldan, I have documented the migration steps taken so far in the [File System Access article](https://web.dev/file-system-access/#accessing-storage-foundation-api-files-from-the-origin-private-file-system) and have also added an [aside](https://web.dev/storage-foundation/#:~:text=There%20is%20an%20ongoing%20effort%20to%20bring%20the%20Storage%20Foundation%20API%20closer%20to%20the%20origin%20private%20file%20system%20of%20the%20File%20System%20Access%20API.%20For%20more%20information%2C%20read%20Accessing%20Storage%20Foundation%20API%20files%20from%20the%20origin%20private%20file%20system.) to the storage foundation article.

**Comment by hcldan** [comment]
@tomayac Thank you for adding these notes, one major area of concern is that it seems a short-cut was taken to reduce the synchronous api access access that used to be available from StorageFoundation apis.  We have already started working and planning for using these apis in workers from WASM code.  

The current WASM FS implementation is not performant enough and have very undesirable consequences for large files.
We saw StorageFoundation sync apis in the workers as a solution to this, but now there will be no way to synchronously do a file operation from a sync JS callout.  This is a huge breakage for us, I urge you to address this issue in the migration (by providing sync parity with async on the api in workers, or by continuing support for StorageFoundation as an alternative).  I would prefer the former.
